Adjust kdump memory if QEMURAM < 2GB and also soft-fail for aarch64

Zaoliang opened the pull request:
we have also the issue on aarch64, see bsc#1161421
see https://progress.opensuse.org/issues/63772
Update memory requirements for kdump on ppc64le64
Set memory to 320 if the SUT has only 1GB of ram
Show soft failure if the SUT is aarch64 too
test verifications:
https://openqa.suse.de/tests/4028997#next_previous (QEMURAM 1024 aarch64)
https://openqa.suse.de/tests/4019082#next_previous (QEMURAM 4086 aarch64)
https://openqa.suse.de/tests/4028977#next_previous (QEMURAM 1024 ppc64le)

okurz commented:
s/softfail/softffail/

okurz commented:
As you extend the reference to the ppc64le bug to aarch64 then please make sure this is also properly described in the bug itself. https://bugzilla.suse.com/show_bug.cgi?id=1161421#c0 mentions that the "tests for aarch64 work fine".

czerw commented:
Please do verification on job which has only 1GB RAM. You may face bsc#1158540 on system with 1G RAM and kdump memory set to 640M.

czerw commented:
> As you extend the reference to the ppc64le bug to aarch64 then please make sure this is also properly described in the bug itself. https://bugzilla.suse.com/show_bug.cgi?id=1161421#c0 mentions that the "tests for aarch64 work fine".

It was only with lower SUT memory. I updated mentioned bsc#1161421.

Zaoliang commented:
> > As you extend the reference to the ppc64le bug to aarch64 then please make sure this is also properly described in the bug itself. https://bugzilla.suse.com/show_bug.cgi?id=1161421#c0 mentions that the "tests for aarch64 work fine".
> 
> It was only with lower SUT memory. I updated mentioned bsc#1161421.

No, this happened sporadic also on worker with RAM 4096: see https://openqa.suse.de/tests/4019164#step/kdump_and_crash/73

Other example is reboot which didn't made. I would say this is performance issue on aarch64 workers on osd.

Zaoliang commented:
https://openqa.suse.de/tests/4019082#next_previous shows for 51 test runs:

1 failure 
1 failure with reboot which is clearly performance issue.

czerw commented:
> No, this happened sporadic also on worker with RAM 4096: see https://openqa.suse.de/tests/4019164#step/kdump_and_crash/73

I just wanted to say, that i didn't observer this issue on aarch64 with 1G RAM that time.

czerw commented:
> https://openqa.suse.de/tests/4019082#next_previous shows for 51 test runs:
> 
> 1 failure
> 1 failure with reboot which is clearly performance issue.

OK. please do verification by cloning this job https://openqa.suse.de/tests/4017488

okurz commented on lib/kdump_utils.pm:
```diff
@@ -121,18 +121,18 @@ sub activate_kdump {
     # get kdump memory size bsc#1161421
     my $memory_total = script_output('kdumptool  calibrate | awk \'/Total:/ {print $2}\'');
-    my $memory_kdump = $memory_total >= 2048 ? 1024 : 640;
+    my $memory_kdump = $memory_total >= 2048 ? 1024 : 320;
```
This change should not be mixed with "Add softfail … on aarch64", better split out

czerw replied:
And I think that 320 is to small for ppc64le, please do verification for this arch  https://openqa.suse.de/tests/4031133#

Zaoliang replied:
> And I think that 320 is to small for ppc64le, please do verification for this arch https://openqa.suse.de/tests/4031133#

please check already provided verification:
https://openqa.suse.de/tests/4028977#next_previous (QEMURAM 1024 ppc64le)

Zaoliang replied:
> This change should not be mixed with "Add softfail … on aarch64", better split out

I don't see the real reason for split out here. If this is not working, yes, I should handle ppc64le and aarch64 differently. but please look at verification runs. I checked them and tried with different sizes.

Zaoliang replied:
320 MB is safe and works well for both when QEMURAM=1024.

czerw replied:
Ok, thanks for the verification.

Zaoliang commented:
> Commits should not lie about what they change. If you say you add a soft fail for one Arch and change the value for a different Arch this is something different. Just split out a comment using git amend or git GUI or git rebase --interactive. Should have taken you less time than writing a comment here :)

updated now. no "lies" anymore...

okurz commented:
I am sorry if I have not been clear but each commit should do one thing and not mix very different things. Please split.

SergioAtSUSE commented:
Agreed with @foursixnine and @Soulofdestiny

okurz commented:
It's good to see that you agree with your colleagues :) About what I don't know as you haven't written anything here.
